feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-base.component.ts

```diff
@@ -146,16 +146,26 @@ export class ConfiguratorAttributeBaseComponent {
    * @param expMode - Is expert mode set?
    * @param label - value label
    * @param techName - value technical name
+   * @param value - item Object which contains the valuePrice
+   * @param isSelected - is item in dropdown list selected
    */
   getLabel(
     expMode: boolean,
     label: string | undefined,
-    techName: string | undefined
+    techName: string | undefined,
+    value?: Configurator.Value
   ): string {
     let title = label ? label : '';
     if (expMode && techName) {
       title += ` / [${techName}]`;
     }
+    if (value && !value.selected) {
```

Comment:
Is mor readable:

>  if (!value?.selected && value?.valuePrice?.value) {
>       if (value?.valuePrice.value < 0) {
>         title += ` [${value.valuePrice?.formattedValue}]`;
>       } else {
>         title += ` [+${value.valuePrice?.formattedValue}]`;
>       }
>     }

Reply:
The tests are still running green

Reply:
Plz, extract into separate method, for instance `getValuePrice(value: Configurator.Value | undefined)`

Reply:
done, just made a few small adaptions: 

changed the order for : 
if (!value.selected && value.valuePrice?.value) {
to
 if (value?.valuePrice?.value && !value.selected) {

also had to make sure that value is not undefined by changing It to "value?" 
the order change was necessary since otherwise for the first statement value.selected value could be possibly undefined